File: iosApp/iosApp/Modules/Authenticate ✅/AuthenticateViewModel.swift
// Created by web3d3v on 11/02/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import Foundation

struct AuthenticateViewModel {
    let title: String
    let password: String
    let passType: PassType
    let passwordPlaceholder: String
    let salt: String
    let saltPlaceholder: String
    let needsPassword: Bool
    let needsSalt: Bool
    
    enum PassType {
        case pin
        case pass
    }
}

File: web3walletcore/src/commonMain/kotlin/com/sonsofcrypto/web3walletcore/modules/authenticate/AuthenticateViewModel.kt
package com.sonsofcrypto.web3walletcore.modules.authenticate

data class AuthenticateViewModel(
    val title: String,
    val password: String,
    val passType: PassType,
    val passwordPlaceholder: String,
    val salt: String,
    val saltPlaceholder: String,
    val needsPassword: Boolean,
    val needsSalt: Boolean,
) {
    enum class PassType { PIN, PASS }
}

